lib/biaser.js

```diff
@@ -0,0 +1,217 @@
+let assert = require('assert');
+let debug = require('debug')('aws-provisioner2:biaser');
+
+/**
+ * A biaser is an object which understands how much we should multiply the
+ * calculated capacity unit price by to determine the price for comparison.
+ * A calculated capacity unit price is the price that we compute by dividing
+ * the price that amazon charges for that instance type by the utility factor
+ * for that instance type in the workerType definition.
+ */
+function Biaser (cfg) {
+  // Store a taskcluster-base.stats Influx object that the queries will be run
+  // against
+  assert(cfg.influx);
+  this.influx = cfg.influx;
+
+  // Max bias age is how long we should keep holding onto bias tables when we
+  // can't fetch new tables
+  assert(cfg.maxBiasAge);
+  assert(typeof cfg.maxBiasAge === 'number');
+  this.__maxBiasAge = cfg.maxBiasAge;
+
+  // Stats age is the string that forms part of an InfluxDB query that
+  // specifies how long of a period that we should consider when we fetch data
+  // for finding bias values
+  assert(cfg.statsAge);
+  assert(typeof cfg.statsAge === 'string');
+  this.__statsAge = cfg.statsAge;
+
+  // When we have kills and spawns, we take the percentage of killed jobs and
+  // multiply it by this number to determine the bias.  Example: 16 kills, 233
+  // spawns is roughly 0.06.  If we set this multipler to 2 we have a bias of
+  // 1.12
+  assert(cfg.killRateMultiplier);
+  assert(typeof cfg.killRateMultiplier === 'number');
+  this.__killRateMultiplier = cfg.killRateMultiplier;
+
+  // We want to give preference to regions where we haven't spawned anything or
+  // had any kills as a way to distribute load
+  assert(cfg.emptyComboBias);
+  assert(typeof cfg.emptyComboBias === 'number');
+  this.__emptyComboBias = cfg.emptyComboBias;
+
+  // Store an empty bias table
+  this.__biasTable = {};
```

Comment:
You should document the table layout... ideally in a comment here.

Note: I think the current design is particularly bad. Instead of `this.__biasTable[region][zone][type]` do something  like `this.__biasTable[region+ '/' + zone + '/' + type]` and you won't have to deal with missing intermediate objects.
But most likely this isn't called very often, so you would probably be fine with an array where `region`, `zone` and `type` are properties on elements of the array.

Nested objects is complicated to code with... and invites too many bugs, if statements and nested loops...